feat(Text): add align prop

bmdalex opened the pull request:
## feat(Text): add textAlign prop

## BREAKING CHANGE MITIGATION

`Header`'s `textAlign` prop values changed from `'left' | 'center' | 'right' | 'justified'` to `'start' | 'end' | 'center' | 'justified'`

**mitigation**: change `textAlign="left"` to `textAlign="start"` and `textAlign="right"` to `textAlign="end"`

### Description

This PR adds `textAlign` prop to the `Text` component in order to allow users to align content to the left, right, center or have it justified.

bmdalex committed:
feat(Text): add textAlign prop

bmdalex committed:
changelog

alinais commented on packages/react/src/themes/base/components/Text/textStyles.ts:
```diff
@@ -46,21 +39,11 @@ export default {
+      ...(weight === 'semilight' && { fontWeight: v.fontWeightSemilight }),
+      ...(weight === 'regular' && { fontWeight: v.fontWeightRegular }),
+      ...(weight === 'semibold' && { fontWeight: v.fontWeightSemibold }),
+      ...(weight === 'bold' && { fontWeight: v.fontWeightBold }),
```
nice visual change

bmdalex committed:
removed wrapper from header examples

mnajdova commented on docs/src/examples/components/Text/Variations/index.tsx:
```diff
@@ -4,6 +4,18 @@ import ExampleSection from 'docs/src/components/ComponentDoc/ExampleSection'
+        <>
+          Text may be aligned to the left, right, center or be justified.
+          <br />
+          You need to use Text as a block element (<code>as='div'</code>) so that textAlign prop can
```
Probably you already considered this, but why don't we just add `display: 'block'` in the styles if the textAlign prop is provided?

bmdalex replied:
Good that you opened this conversation. I have considered it briefly but thought it might be unexpected behavior for the user.

Giving it a second thought if a user adds `textAlign` they probably expect the `Text` component to behave as a block otherwise alignment does not make sense. So I think you're right here. Let me make the change and ask others to chip in as well: @layershifter @kuzhelov @miroslavstastny

bmdalex replied:
applied `display: 'block' ` when `textAlign` prop is provided

bmdalex committed:
addressed PR comments

mnajdova reviewed (approved):
LGTM, nice cleanups too! 👍

miroslavstastny commented on packages/react/src/components/Header/Header.tsx:
```diff
@@ -36,7 +37,7 @@ export interface HeaderProps
   description?: ShorthandValue<HeaderDescriptionProps>
 
   /** Align header content. */
-  textAlign?: 'left' | 'center' | 'right' | 'justified'
```
My main question is whether we want to have a design term for alignment (like we had here previously) or expose a CSS property like you are doing in this PR.

My opinion is that we should stick with a design term, for that reason `left, center, right, justified` work for me better that all the valid CSS values (`-moz-initial` is definitely not a design term, if you need it, use styles).
And I would even think about replacing `left` and `right` with `start` and `end` to comply with our direction-agnostic terminology.

mnajdova replied:
I agree with @miroslavstastny points. I guess @Bugaa92 was just following the way it is already implemented in the `Header` component. Maybe the best option would be to introduce a different prop (something that does not necessarily correlate directly to css prop), and provide the mention values there as options.

bmdalex replied:
@miroslavstastny - very good points, I'm confident others agree; I see @mnajdova liked the message so 3 votes already. 

The only issue I have with our `start` and `end` as values of `textAlign` prop is that they mean something different for the actual `textAlign` CSS prop.

So when we'll have

```jsx
<Text textAlign="end" .. />
```
 will we render
```html
<span style="text-align:right" ../>
```
or
```html
<span style="text-align:end" ../>
```
?

These have different functionality in RTL mode as you can see here:
https://codesandbox.io/s/scrollableelement-wm5tb

![Screenshot 2019-07-17 at 16 51 01](https://user-images.githubusercontent.com/5442794/61385662-1b2c2100-a8b3-11e9-805d-fc5e970a452a.png)

The options I see right now are:

**1. have `textAlign` prop with values `'start' | 'end' | 'center' | 'justified'`** and forward values as they are to CSS `textAlign`

**2. have `textAlign` prop with values `'start' | 'end' | 'center' | 'justified'`** and:
- forward values as they are to CSS `textAlign` for `center` and `justified`
- convert `start` and `end` prop values to `left` and `right` CSS values

**3. keep `textAlign` prop with values `'left' | 'right' | 'center' | 'justified'`** (as it is now for `Header`) and forward values as they are to CSS `textAlign`

**4. have `textAlign` prop with values `'before' | 'after' | 'center' | 'justified'`** and:
- forward values as they are to CSS `textAlign` for `center` and `justified`
- convert `before` and `after` prop values to `left` and `right` CSS values

@mnajdova @miroslavstastny what's your take on all this?

miroslavstastny replied:
FWIW in your codesandbox example you are not using Stardust `Provider`, with that the results are different in RTL:
![image](https://user-images.githubusercontent.com/9615899/61387423-6136b400-a8b6-11e9-9345-fd72fdcf29db.png)

#### Source
```jsx
  <>
  LTR Provider
  <Provider styles={{border: '2px dashed red', padding: '20px'}}>
      <Text textAlign="right">Test align right</Text>
      <Text textAlign="end">Test align end</Text>
  </Provider>
  <br/>
  RTL Provider
  <Provider rtl styles={{border: '2px dashed red', padding: '20px'}}>
      <Text textAlign="right">Test align right</Text>
      <Text textAlign="end">Test align end</Text>
  </Provider>
  </>
```

So this makes `end` CSS value useless in my eyes.

I would go with option 2 and, as @mnajdova suggests, think about renaming textAlign prop to mentally detach it from the CSS prop.

bmdalex replied:
> I would go with option 2

done

>  as @mnajdova suggests, think about renaming textAlign prop 

what about having it just `align`? we already have it for other components (`Menu`, `Dropdown`, `FlexItem`) and even though it has different meaning in the other component I think it's obvious what the meaning is for `Text` and `Header`.

So, what do you guys think about renaming `textAlign` prop to `align`? @mnajdova @miroslavstastny

mnajdova replied:
Agree, I vote for `align` too

bmdalex replied:
done, thanks guys!

bmdalex committed:
addressed PR comments 2:
- changed textAlign prop possible values;
- added method to convert textAlign prop value to CSS prop value

bmdalex committed:
converted textAlign prop to align for Text and Header

bmdalex committed:
justified -> justify
